--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ find_package(Threads REQUIRED)
 
 set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+include("GNUInstallDirs")
 
 if (DISABLE_AUTO_INGEST)
   add_definitions(-DDISABLE_AUTO_INGEST)
@@ -130,4 +131,5 @@ if (NOT DISABLE_FTL_APP)
 endif()
 
 # Install rules
-install(TARGETS ftl DESTINATION lib)
\ No newline at end of file
+install(TARGETS ftl DESTINATION
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
\ No newline at end of file
